dask_planner/src/sql.rs

```diff
@@ -148,20 +148,20 @@ impl ContextProvider for DaskSQLContext {
 
         match name {
             "year" => {
-                let sig = Signature::variadic(vec![DataType::Int64], Volatility::Immutable);
+                let sig = Signature::exact(vec![DataType::Int64], Volatility::Immutable);
```

Comment:
Interested in how this would work out with non `Int64` integers - I think we would get some warnings raised by the optimizer, but ultimately not end up doing extra casting operations on the Python side because the types are considered similar by `similar_type`.

Might be worth opening an issue to see if we can suppress optimizer warnings by default, as they come up quite a lot during normal use cases.